std/recursion/plonk/verifier.go

```diff
@@ -1242,33 +1241,128 @@ func (v *Verifier[FR, G1El, G2El, GtEl]) SwitchVerificationKey(bvk BaseVerifying
 			return ret, fmt.Errorf("inconsistent circuit verification key")
 		}
 	}
-	for i := len(cvks); i < 4; i++ {
-		cvks = append(cvks, ret.CircuitVerifyingKey)
-	}
 	cvk := CircuitVerifyingKey[FR, G1El]{
 		Qcp:                         make([]kzg.Commitment[G1El], nbCmts),
 		CommitmentConstraintIndexes: make([]frontend.Variable, nbCmts),
 	}
-	bIdx := bits.ToBinary(v.api, idx, bits.WithNbDigits(2))
-	cvk.Size = v.api.Lookup2(bIdx[0], bIdx[1], cvks[0].Size, cvks[1].Size, cvks[2].Size, cvks[3].Size)
-	cvk.SizeInv = *v.scalarApi.Lookup2(bIdx[0], bIdx[1], &cvks[0].SizeInv, &cvks[1].SizeInv, &cvks[2].SizeInv, &cvks[3].SizeInv)
-	cvk.Generator = *v.scalarApi.Lookup2(bIdx[0], bIdx[1], &cvks[0].Generator, &cvks[1].Generator, &cvks[2].Generator, &cvks[3].Generator)
-	for i := range cvk.S {
-		cvk.S[i].G1El = *v.curve.Lookup2(bIdx[0], bIdx[1], &cvks[0].S[i].G1El, &cvks[1].S[i].G1El, &cvks[2].S[i].G1El, &cvks[3].S[i].G1El)
-	}
-	cvk.Ql.G1El = *v.curve.Lookup2(bIdx[0], bIdx[1], &cvks[0].Ql.G1El, &cvks[1].Ql.G1El, &cvks[2].Ql.G1El, &cvks[3].Ql.G1El)
-	cvk.Qr.G1El = *v.curve.Lookup2(bIdx[0], bIdx[1], &cvks[0].Qr.G1El, &cvks[1].Qr.G1El, &cvks[2].Qr.G1El, &cvks[3].Qr.G1El)
-	cvk.Qm.G1El = *v.curve.Lookup2(bIdx[0], bIdx[1], &cvks[0].Qm.G1El, &cvks[1].Qm.G1El, &cvks[2].Qm.G1El, &cvks[3].Qm.G1El)
-	cvk.Qo.G1El = *v.curve.Lookup2(bIdx[0], bIdx[1], &cvks[0].Qo.G1El, &cvks[1].Qo.G1El, &cvks[2].Qo.G1El, &cvks[3].Qo.G1El)
-	cvk.Qk.G1El = *v.curve.Lookup2(bIdx[0], bIdx[1], &cvks[0].Qk.G1El, &cvks[1].Qk.G1El, &cvks[2].Qk.G1El, &cvks[3].Qk.G1El)
-	for i := range cvk.Qcp {
-		cvk.Qcp[i].G1El = *v.curve.Lookup2(bIdx[0], bIdx[1], &cvks[0].Qcp[i].G1El, &cvks[1].Qcp[i].G1El, &cvks[2].Qcp[i].G1El, &cvks[3].Qcp[i].G1El)
-	}
-	for i := range cvk.CommitmentConstraintIndexes {
-		cvk.CommitmentConstraintIndexes[i] = v.api.Lookup2(bIdx[0], bIdx[1], cvks[0].CommitmentConstraintIndexes[i], cvks[1].CommitmentConstraintIndexes[i], cvks[2].CommitmentConstraintIndexes[i], cvks[3].CommitmentConstraintIndexes[i])
-	}
+	v.muxCircuitVerifyingKey(idx, cvks, &cvk)
 	return VerifyingKey[FR, G1El, G2El]{
 		BaseVerifyingKey:    bvk,
 		CircuitVerifyingKey: cvk,
 	}, nil
 }
+
+func (v *Verifier[FR, G1El, G2El, GtEl]) muxCircuitVerifyingKey(idx frontend.Variable, cvks []CircuitVerifyingKey[FR, G1El], ret *CircuitVerifyingKey[FR, G1El]) {
+	var cvkT CircuitVerifyingKey[FR, G1El]
```

Comment:
so, discovering this, I'm a bit confused as to what is the purpose of the method and why `reflect` is a good choice here. Can you add couple of lines in comments to contextualise this?